modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/persistence/store/PageStore.java

```diff
@@ -0,0 +1,156 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.pagememory.persistence.store;
+
+import java.io.Closeable;
+import java.nio.ByteBuffer;
+import org.apache.ignite.lang.IgniteInternalCheckedException;
+
+/**
+ * Persistent store of pages.
+ */
+public interface PageStore extends Closeable {
+    /** Type for affinity partitions. */
+    byte TYPE_DATA = 1;
+
+    /** Type for index partition. */
+    byte TYPE_IDX = 2;
+
+    /**
+     * Adds page write listener.
+     *
+     * @param listener Page write listener.
+     */
+    void addWriteListener(PageWriteListener listener);
+
+    /**
+     * Removes page write listener.
+     *
+     * @param listener Page write listener.
+     */
+    void removeWriteListener(PageWriteListener listener);
+
+    /**
+     * Returns {@code true} if the page store exists.
+     */
+    boolean exists();
+
+    /**
+     * Allocates next page index.
+     *
+     * @return Next page index.
+     * @throws IgniteInternalCheckedException If failed to allocate.
+     */
+    long allocatePage() throws IgniteInternalCheckedException;
+
+    /**
+     * Returns number of allocated pages.
+     */
+    long pages();
+
+    /**
+     * Reads a page.
+     *
+     * @param pageId Page ID.
+     * @param pageBuf Page buffer to read into.
+     * @param keepCrc By default, reading zeroes CRC which was on page store, but you can keep it in {@code pageBuf} if set {@code
+     * keepCrc}.
+     * @return {@code True} if page has been read successfully, {@code false} if page hasn't been written yet.
+     * @throws IgniteInternalCheckedException If reading failed (IO error occurred).
+     */
+    boolean read(long pageId, ByteBuffer pageBuf, boolean keepCrc) throws IgniteInternalCheckedException;
+
+    /**
+     * Reads a page store header.
+     *
+     * @param buf Buffer to write to.
+     * @throws IgniteInternalCheckedException If failed.
+     */
+    void readHeader(ByteBuffer buf) throws IgniteInternalCheckedException;
+
+    /**
+     * Writes a page.
+     *
+     * @param pageId Page ID.
+     * @param pageBuf Page buffer to write from.
+     * @param tag Partition page store version, 1-based incrementing counter. For outdated pages {@code tag} has lower value,
+     *      and write does nothing.
+     * @param calculateCrc If {@code false} crc calculation will be forcibly skipped.
+     * @throws IgniteInternalCheckedException If page writing failed (IO error occurred).
+     */
+    void write(long pageId, ByteBuffer pageBuf, int tag, boolean calculateCrc) throws IgniteInternalCheckedException;
+
+    /**
+     * Returns page offset within the page store.
+     *
+     * @param pageId Page ID.
+     */
+    long pageOffset(long pageId);
```

Comment:
A little later it will use **org.apache.ignite.internal.processors.cache.persistence.file.FilePageStore** in method **FilePageStore#read** and **FilePageStore#write**, I can remove it from the public api, wdyt?

Reply:
Remove it